### ISSUE TITLE
Add enhanced user tracing during the restore procedure

### ISSUE THREAD

**Issue by dasistwas** [Wunderbyte-GmbH/moodle-local_remote_backup_provider#3]
There has to be an analyses of users that are included in the backup and how they are going to be handled during the restore process. So before starting the restore process, there should be a check for users, displaying the result and separate users in different categories:

1. Users which are existing on  the remote platform that match users on the local platform
2. Users which are existing on the remote platform and who are going to be created as new users on the local platform (if the user who is performing the restore has the appropriate rights to create new users)
3. Users with problems. Users with same useranme but different email adresses. Or special admin users. Admin users with different email will make restore with userdata impossible. As of now Moodle does not provide a way to solve that problem, but only provides the possibility to abort the restore process.

- [x] Before restore uncompress backup file to a temp folder
- [x] Check the user list
- [x] Perform comparison
- [x] Display user list
- [x] Provide action buttons: Download users, Proceed with restore, Cancel restore

**Comment by dasistwas** [comment]
User check is done there:
https://github.com/moodle/moodle/blob/e69a3f20ee1604fb45cc23cbd436121c720adf97/backup/util/dbops/restore_dbops.class.php#L1538

**Comment by dasistwas** [comment]
Also helpful to look at that:
https://github.com/moodle/moodle/blob/e69a3f20ee1604fb45cc23cbd436121c720adf97/backup/util/dbops/restore_dbops.class.php#L1128

**Comment by dasistwas** [comment]
We need probably that restore process already done there:
https://github.com/Wunderbyte-GmbH/moodle-mod_mooduell/commit/ed271f8d435ecb0fe72f2535f876da337246e844

**Comment by dasistwas** [comment]
Issues:

- [x] Readability of colors

- [x] Actions are unclear

- [x] Issue message ist misleading
